onnx/optimizer/passes/fuse_consecutive_transposes.h

```diff
@@ -28,10 +28,10 @@ struct FuseConsecutiveTransposes final : public PredicateBasedPass {
     std::vector<int64_t> ret;
     ret.reserve(t1.size());
     for (size_t i = 0; i < t1.size(); i++) {
-      ONNX_ASSERT(t1[i] < static_cast<int64_t>(t2.size()));
+      ONNX_ASSERT(t2[i] < static_cast<int64_t>(t1.size()));
```

Comment:
do you mind to add some test cases so that this can't be regressed?

Reply:
In the next commit, by doing infer_shapes we can make sure the graph and intput/output tensors are consistent before and after the optimization.  

Without the fix:
```
PASSED optimizer_test.py::TestOptimizer::test_fuse_transpose_default
PASSED optimizer_test.py::TestOptimizer::test_fuse_transpose_default_no_fuse
PASSED optimizer_test.py::TestOptimizer::test_fuse_transpose_into_gemm
FAILED optimizer_test.py::TestOptimizer::test_fuse_transpose - RuntimeError: Inferred shape and existing shape differ in dimension 0: (4) vs (2)
```

With the fix:
```
PASSED optimizer_test.py::TestOptimizer::test_fuse_transpose
PASSED optimizer_test.py::TestOptimizer::test_fuse_transpose_default
PASSED optimizer_test.py::TestOptimizer::test_fuse_transpose_default_no_fuse
PASSED optimizer_test.py::TestOptimizer::test_fuse_transpose_into_gemm
```